lib/src/main/java/com/solus/sdk/noebscall/NOEBSClient.java

```diff
@@ -127,9 +150,21 @@ private String getKey() {
                 return errorBaseResponse;
             } else if (response.code() == 404) {
                 String responseBodey = response.body().string();
+
                 ErrorResponse errorResponse = mapper.readValue(responseBodey, ErrorResponse.class);
                 errorBaseResponse.setResponse(errorResponse);
                 return errorBaseResponse;
+            }else if (response.code() == 502) { // for ebs error messages
+                String responseBodey = response.body().string();
+
+                JsonObject jsonObject = gson.fromJson(responseBodey, JsonObject.class);
+
+                logger.info("gson response " + jsonObject.get("details").toString());
+                logger.info("ebs_error_502 is: " + responseBodey);
+                ErrorResponse errorResponse = mapper.readValue(jsonObject.get("details").toString(), ErrorResponse.class);
```

Comment:
**pass responsebody to readvalue func  instead of jsonObject**

Reply:
On it boss